File: Data Structures & Algorithms/products-of-array-discluding-self/submission-4.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int pre=1,pos=1;
        int res[]=new int[nums.length];
        Arrays.fill(res,1);
        for(int i=0;i<nums.length;i++){
            res[i]=res[i]*pre;
            pre=nums[i]*pre;
        }
        for(int i=nums.length-1;i>=0;i--){
            res[i]=res[i]*pos;
            pos=nums[i]*pos;
        }
        return res;
    }
}  

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-4.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        i,j=0,len(nums)-1
        while i<j:
            mid=(j-i)//2+i
            if nums[mid]>=nums[j]:
                i=mid+1
            else:
                j=mid
        return nums[i]